src/db/connection.js

```diff
@@ -32,7 +32,8 @@ const clearDatabase = async () => {
   const results = [];
   /* eslint-disable no-restricted-syntax, guard-for-in */
   for (const key in collections) {
-    results.push(mongoose.connection.dropCollection(key));
+    // results.push(mongoose.connection.dropCollection(key));
+    results.push(collections[key].deleteMany());
```

Comment:
@Shrreya I just want to add a note here, we spent the entire day yesterday trying to solve some error. There was an error saying `MongoError: ns not found` and after a lot of testing I found that it is better to delete what is inside the collection rather than dropping the collection, because this error is occurring due to trying to drop a collection that does not exist,  the test is running fine in the first test file, but when the second test file starts running we are calling this function again so it comes here and try to drop the collections that are already dropped in the previous test (I don't know why but keys are still existing inside the `cellections` variable even though we dropped them).

Reply:
and because this error is thrown, the `afterAll` function is not running everything inside it so `server.close()` is not being called and another error is occurring that says `port already in use`
If we leave this function like this, we will get rid of these two errors

Reply:
@AmmarBaki2 Thank you for letting me know! Looks like you had an intense debugging session, but happy to know you found the issue and resolved it.

It's a good learning that `deleteMany()` does a cleaner job than `dropCollection()`.